src/ints/mouse.cpp

```diff
@@ -18,1549 +18,932 @@
+    case 2: return MouseEventId::ReleasedMiddle;
+    case 3:
+    case 4:
+        return changed_12S ? MouseEventId::ReleasedMiddle
+                           : MouseEventId::NotDosEvent;
+    default: return MouseEventId::NotDosEvent;
+    }
 }
 
 // ***************************************************************************
 // External notifications
 // ***************************************************************************
 
-void MOUSE_SetSensitivity(const int32_t sensitivity_x, const int32_t sensitivity_y)
+void MOUSE_SetConfig(const bool new_raw_input)
 {
-	auto adapt = [](const int32_t sensitivity) {
-		const float tmp = std::clamp(static_cast<float>(sensitivity) / 100.0f,
-		                             -100.0f,
-		                             100.0f);
-
-		if (tmp >= 0)
-			return std::max(tmp, 0.01f);
-		else
-			return std::min(tmp, -0.01f);
-	};
-
-	mouse_config.sensitivity_x = adapt(sensitivity_x);
-	mouse_config.sensitivity_y = adapt(sensitivity_y);
+    raw_input = new_raw_input;
 }
 
 void MOUSE_NewScreenParams(const uint16_t clip_x, const uint16_t clip_y,
                            const uint16_t res_x, const uint16_t res_y,
                            const bool fullscreen, const uint16_t x_abs,
                            const uint16_t y_abs)
 {
-	mouse_video.clip_x = clip_x;
-	mouse_video.clip_y = clip_y;
+    mouse_video.clip_x = clip_x;
+    mouse_video.clip_y = clip_y;
 
-	// Protection against strange window sizes,
-	// to prevent division by 0 in some places
-	mouse_video.res_x = std::max(res_x, static_cast<uint16_t>(2));
-	mouse_video.res_y = std::max(res_y, static_cast<uint16_t>(2));
+    // Protection against strange window sizes,
+    // to prevent division by 0 in some places
+    mouse_video.res_x = std::max(res_x, static_cast<uint16_t>(2));
+    mouse_video.res_y = std::max(res_y, static_cast<uint16_t>(2));
 
-	mouse_video.fullscreen = fullscreen;
+    mouse_video.fullscreen = fullscreen;
 
-	MOUSEVMWARE_NewScreenParams(x_abs, y_abs);
+    MOUSEVMM_NewScreenParams(x_abs, y_abs);
+    MOUSE_NotifyStateChanged();
 }
 
-void MOUSE_EventMoved(const int16_t x_rel, const int16_t y_rel, const uint16_t x_abs,
-                      const uint16_t y_abs, const bool is_captured)
+void MOUSE_NotifyRateDOS(const uint8_t rate_hz)
+{
+    auto &val_mov = queue.start_delay_dos_mov;
+    auto &val_btn = queue.start_delay_dos_btn;
+
+    // Convert rate in Hz to delay in milliseconds
+    val_mov = static_cast<uint8_t>(1000 / rate_hz);
+    // Cheat a little, do not allow to set more than 10 milliseconds. Some
+    // old games might have tried to set lower rate to reduce number of IRQs
+    // and save CPU power - this is no longer necessary. Windows 3.1 also
+    // sets a suboptimal value in its PS/2 driver.
+    // Original DOSBox forces 5 milliseconds all the time, but this is
+    // probably an overkill - 10 milliseconds is already twice the typical
+    // 50 Hz screen refresh rate.
```

Comment:
Just pointing out that for standard VGA modes the refresh rate is 70Hz, and 60Hz for EGA. Not sure if that changes anything, though.

Reply:
Updated comment, changed it a little to have 2 readings per frame even at 70 Hz.